jdaviz/core/template_mixin.py

```diff
@@ -230,32 +272,46 @@ def _update_has_subregions(self):
     def labels(self):
         return [s['label'] for s in self.items if 'label' in s.keys()]
 
+    @cached_property
+    def selected_item(self):
+        for item in self.items:
+            if item['label'] == self.selected:
+                return item
+
     @cached_property
     def selected_obj(self):
-        if self.selected == "Entire Spectrum":
+        if self.selected == self._default_text:
             return None
-        return self.app.get_subsets_from_viewer("spectrum-viewer",
-                                                subset_type="spectral").get(self.selected)
+        subset_type = self.selected_item['type']
+        for viewer in ['spectrum-viewer', 'flux-viewer']:
+            match = self.app.get_subsets_from_viewer(viewer,
+                                                     subset_type=subset_type).get(self.selected)
+            if match is not None:
+                return match
```

Comment:
@pllim - is this equivalent to the logic currently in aperture photometry's `_get_region_from_subset`?  Should that logic move here or should aperture photometry make use of this property instead of having its own logic?

Cleaning up this logic will likely help get the aperture photometry tests passing again.

Reply:
I am not sure yet. I need to refactor aperture photometry logic for big mosaic image, which means I will try to only access the array values around the aperture. I have not yet investigated if this method grabs the whole array or only what I want. I will know more after I am done with #1138 .

Reply:
what do you think would be the best path forward then?  Leave the logic as is, get the tests passing, and then handle any changes in 1138 (preferably moving any general logic into the new component)?  Or would you rather I put this PR on hold until you have that work done?

Reply:
Is this PR going to the Mar 25 release? My answer would depend on that.

Reply:
Or you can just apply to Cubeviz for now and we can worry about Imviz later. I think this PR is a huge plus for Cubeviz, which can have both spectral and spatial subsets, but not so for Imviz.